### PR TITLE
[SPARK-45110][BUILD] Upgrade rocksdbjni to 8.8.1

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to upgrade rocksdbjni from `8.3.2` to `8.8.1`.

Why version `8.8.1`? Because so far, `32` tests have been conducted based on version `8.6.7` or `8.8.1`, and no previous core issues have been found. The later versions have not been rigorously validated.

### Why are the changes needed?
1.The full release notes: 
- https://github.com/facebook/rocksdb/releases/tag/v8.8.1
- https://github.com/facebook/rocksdb/releases/tag/v8.7.3
- https://github.com/facebook/rocksdb/releases/tag/v8.6.7
- https://github.com/facebook/rocksdb/releases/tag/v8.5.4
- https://github.com/facebook/rocksdb/releases/tag/v8.5.3
- https://github.com/facebook/rocksdb/releases/tag/v8.4.4
- https://github.com/facebook/rocksdb/releases/tag/v8.3.3

2.Bug Fixes, eg:
- Fixed a bug where compaction read under non direct IO still falls back to RocksDB internal prefetching after file system's prefetching returns non-OK status other than Status::NotSupported()
- Fix a bug with atomic_flush=true that can cause DB to stuck after a flush fails
- Fix a bug where if there is an error reading from offset 0 of a file from L1+ and that the file is not the first file in the sorted run, data can be lost in compaction and read/scan can return incorrect results.
- Fix a bug where iterator may return incorrect result for DeleteRange() users if there was an error reading from a file.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Pass GA.
- Manually test:
```
./build/mvn clean install -pl core -am -Dtest.exclude.tags=org.apache.spark.tags.ExtendedLevelDBTest -fn

```

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
This PR will synchronize the Master every day and run twice a day to verify Core Crash issues.

**Comment by panbingkun** [comment]
- 20231125
https://github.com/panbingkun/spark/actions/runs/6987302202/job/19014769999
RocksDBStateStoreStreamingAggregationSuite
image

**Comment by panbingkun** [comment]
> Thank you for keeping tracking this, @panbingkun . How about Today's status?
@dongjoon-hyun 
- From the past `11` test processes, no similar `fatal errors`  (https://github.com/apache/spark/pull/42862) have been found before.
https://github.com/apache/spark/pull/42862#issuecomment-1715986788
image
- Every day, I try to run at least one time after rebasing master, let's continue to observe for a few days.

**Comment by dongjoon-hyun** [comment]
Thank you so much, @panbingkun !

**Comment by panbingkun** [comment]
> hmm... @panbingkun Why is the upgrade target 8.6.7 instead of 8.7.3? Considering the timeline, 8.7.3 was already released when testing began. Are there any known issues with 8.7.3?

When I started this PR, I found that the release on GitHub(`8.7.3`) was inconsistent with the release on Maven Repo(`8.6.7`), so I chose the `8.6.7` version. Today, I checked and now the version has reached `8.8.1`.

File: python/docs/source/user_guide/sql/index.rst [hunk 1]
```diff
@@ -16,9 +16,9 @@
     under the License.
 
 
-=========
-Spark SQL
-=========
+========================
+Spark SQL and DataFrames
```
[the remark this answers is not in the document]
Comment (panbingkun) [reply python/docs/source/user_guide/sql/index.rst:20]: Okay, I mistakenly submitted some irrelevant documents, I will withdraw them immediately.

**Comment by LuciferYang** [comment]
Has this test been completed? Can we upgrade to the new version now?

**Comment by panbingkun** [comment]
> Has this test been completed? Can we upgrade to the new version now?

Let's continue to verify until the end of this week?

**Comment by LuciferYang** [comment]
> > Has this test been completed? Can we upgrade to the new version now?
> 
> Let's continue to verify until the end of this week?

ok

**Comment by LuciferYang** [comment]
Can we merge this one first?

**Comment by panbingkun** [comment]
> Can we merge this one first?

I think we can give it a try.

**Review by srowen — COMMENTED** [review]
Where are we on this one - any reason to believe we shouldn't merge?

**Comment by panbingkun** [comment]
> Where are we on this one - any reason to believe we shouldn't merge?

I think we can merge this version because it has undergone a long period of testing and validation.
Let me rebase the master again and do the final verification.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**